heudiconv/convert.py

```diff
@@ -0,0 +1,394 @@
+        # intension???? XXX
+        outpath = outdir
+        for idx, itemgroup in enumerate(items):
+            if not isinstance(itemgroup, list):
+                itemgroup = [itemgroup]
+            for subindex, item in enumerate(itemgroup):
+                parameters = {}
+                if isinstance(item, dict):
+                    parameters = {k: v for k, v in item.items()}
+                    item = parameters['item']
+                    del parameters['item']
+                # some helper meta-varaibles
+                parameters.update(dict(
+                    item=idx + 1,
+                    subject=subject,
+                    seqitem=item,
+                    subindex=subindex + 1,
+                    session='ses-' + str(ses),
+                    bids_subject_session_prefix=
+                        'sub-%s' % subject + (('_ses-%s' % ses) if ses else ''),
+                    bids_subject_session_dir=
+                        'sub-%s' % subject + (('/ses-%s' % ses) if ses else ''),
+                    # referring_physician_name
+                    # study_description
+                    ))
+                try:
+                    files = filegroup[item]
+                except KeyError:
+                    files = filegroup[(str if PY3 else unicode)(item)]
+                outprefix = template.format(**parameters)
+                convert_info.append((op.join(outpath, outprefix),
+                                    outtype, files))
+    return convert_info
+
+
+def prep_conversion(sid, dicoms, outdir, heuristic, converter, anon_sid,
+                   anon_outdir, with_prov, ses, bids, seqinfo, min_meta,
+                   overwrite):
+    if dicoms:
+        lgr.info("Processing %d dicoms", len(dicoms))
+    elif seqinfo:
+        lgr.info("Processing %d pre-sorted seqinfo entries", len(seqinfo))
+    else:
+        raise ValueError("neither dicoms nor seqinfo dict was provided")
+
+    if bids and not sid.isalnum(): # alphanumeric only
+        sid, old_sid = convert_sid_bids(sid)
+
+    if not anon_sid:
+        anon_sid = sid
+    if not anon_outdir:
+        anon_outdir = outdir
+
+    # Generate heudiconv info folder
+    idir = op.join(outdir, '.heudiconv', sid)
+    if bids and ses:
+        idir = op.join(idir, 'ses-%s' % str(ses))
+    if anon_outdir == outdir:
+        idir = op.join(idir, 'info')
+    if not op.exists(idir):
+        os.makedirs(idir)
+
+    shutil.copy(heuristic.filename, idir)
+    ses_suffix = "_ses-%s" % ses if ses is not None else ""
+    info_file = op.join(idir, '%s%s.auto.txt' % (sid, ses_suffix))
+    edit_file = op.join(idir, '%s%s.edit.txt' % (sid, ses_suffix))
+    filegroup_file = op.join(idir, 'filegroup%s.json' % ses_suffix)
+
+    # MG - maybe add an option to force rerun?
+    # related issue : https://github.com/nipy/heudiconv/issues/84
+    if op.exists(edit_file) and overwrite == True:
```

Comment:
I got to like Pythonic generic programming semantics, so I would have just said `and overwrite`... why bother with `overwrite == True`?

Reply:
whoops...guess i was thinking verbose atm